src/api/fileops/fileOperationsModule.ts

```diff
@@ -0,0 +1,40 @@
+import { ReflectiveInjector } from 'injection-js';
+import { RequestExecuter } from '../../internal/executer';
+import { IModule } from '../core/module';
+import { AuthOptions } from '../core/tokenManager';
+import { FilesetInterface, FilesetName } from './fileops.interfaces';
+import { Fileset } from './fileset';
+import { FileUploader } from './fileUploader';
+
+export class FileOperationsModule<F> implements IModule {
+  private injector: ReflectiveInjector;
+
+  public init(coreInjector: ReflectiveInjector): Promise<this> {
+    this.injector = coreInjector;
+    return Promise.resolve(this);
+  }
+
+  public fileset<T extends object = any>(fileset: string, auth?: string): Fileset<T, FilesetInterface<T>, F> {
```

Comment:
with `extends object = any` do you want to avoid using primitive types?

Reply:
Exactly, <T> has to be an object because it will be expanded by default fileset record fields (such as id, name and size) and could be omitted if a user doesn't want to specify a type or does not have any fields in the fileset